spack/local/packages/palace/package.py

```diff
@@ -22,38 +33,124 @@ class Palace(CMakePackage, CudaPackage, ROCmPackage):
     # Note: 'rocm' and 'amdgpu_target' variants are added by the ROCmPackage
     variant("shared", default=True, description="Build shared libraries")
     variant("int64", default=False, description="Use 64 bit integers")
-    variant("openmp", default=False, description="Use OpenMP for shared-memory parallelism")
     variant(
-        "superlu-dist", default=True, description="Build with SuperLU_DIST sparse direct solver"
+        "openmp", default=False, description="Use OpenMP for shared-memory parallelism"
+    )
+    variant(
+        "superlu-dist",
+        default=True,
+        description="Build with SuperLU_DIST sparse direct solver",
+    )
+    variant(
+        "strumpack",
+        default=False,
+        description="Build with STRUMPACK sparse direct solver",
+    )
+    variant(
+        "sundials",
+        default=True,
+        description="Build with SUNDIALS differential/algebraic equations solver",
     )
-    variant("strumpack", default=False, description="Build with STRUMPACK sparse direct solver")
-    variant("sundials", default=True, description="Build with SUNDIALS differential/algebraic equations solver")
     variant("mumps", default=False, description="Build with MUMPS sparse direct solver")
     variant("slepc", default=True, description="Build with SLEPc eigenvalue solver")
     variant("arpack", default=False, description="Build with ARPACK eigenvalue solver")
-    variant("libxsmm", default=True, description="Build with LIBXSMM backend for libCEED")
     variant("magma", default=True, description="Build with MAGMA backend for libCEED")
+    variant(
+        "libxsmm", default=True, description="Build with libxsmm backend for libCEED"
+    )
     variant(
         "gslib",
         default=True,
         description="Build with GSLIB library for high-order field interpolation",
     )
 
-    # Dependencies
+    ## -- Sparse Direct Solvers --
+    conflicts(
+        "~superlu-dist~strumpack~mumps",
+        msg="Need at least one sparse direct solver",
+    )
+
+    ## -- MUMPS --
+    conflicts("^mumps+int64", msg="Palace requires MUMPS without 64 bit integers")
+    with when("+mumps"):
+        depends_on("mumps+metis+parmetis")
+        depends_on("mumps+shared", when="+shared")
+        depends_on("mumps~shared", when="~shared")
+        depends_on("mumps+openmp", when="+openmp")
+        depends_on("mumps~openmp", when="~openmp")
+
+    ## -- SuperLU-Dist --
+    with when("+superlu-dist"):
+        depends_on("superlu-dist+parmetis~openmp~cuda~rocm")
```

Comment:
```suggestion
        depends_on("superlu-dist+parmetis")
```

Clearly this was missed ◡̈